### DISCUSSION TITLE
X axis displays more values than inputed

### DISCUSSION THREAD

**Discussion by giannism13** [patrykandpatrick/vico#449] [Questions] [answered]
Hello. So I understand that this has been kinda answered before #304 , but I couldn't make the solution suggested there work. So here is my problem:

I'm retrieving from an API objects with a Float value and a date represented in millis since 1/1/1970 contained in a Long.

```kotlin
selectedFieldRecords.forEach { field -> valuesStateList.add(entryOf(field.date, field.value))}
```

`selectedFieldRecords` is a MutableStateList that contains said objects retrieved and `valuesStateList` is list where I'm storing the entries to display.

I've gone ahead and used the code from the documentation to create my entryModel
```kotlin
val data = viewModel.valuesStateList.toList().associate { (dateMillis, yValue) -> dateMillis.toLong() to yValue }
val xValuesToDates = data.keys.associateBy { it }
val entryModelValues = xValuesToDates.keys.zip(data.values, ::entryOf)
val mChartEntryModel = entryModelOf(entryModelValues)
```

I have made a function that converts a date in millis format to String like `29/10/2023 15:31` and I'm using it in the X axis' value formatter as such:
```kotlin
val horizontalAxis = rememberBottomAxis(
	valueFormatter = { value, _ ->
	     DateUtils.getDateTimeForChart(value.toLong())
	},
	itemPlacer = AxisItemPlacer.Horizontal.default(spacing = 2),
	labelRotationDegrees = 90f,
	label = xLabels,
	tickLength = 0.dp,
)

Chart(
	chart = lineChart(),
	model = mChartEntryModel,
	startAxis = verticalAxis,
	bottomAxis = horizontalAxis,
	modifier = Modifier.fillMaxHeight(0.7F)
)
```
But at this point I'm not sure how to normalize the input and while I only have 6 entries to display instead I'm getting thousands.

**Comment by patrickmichalik** [comment]
Hello! Just to confirm, would you like the points to be spaced out proportionally (depending on the deltas between their timestamps)?

**Reply by giannism13** [reply]
From my experience there seems to be zoom out limitations (are there really?) so no.

Update: I came across this #400. Using what you explained there and some chatGPT I got it working. In the future I might need to space the points proportionally; maybe with a scaled down distance too. How would I do that?

**Reply by patrickmichalik** [reply, marked as answer]
I’m glad to hear you’ve found a solution. Scrollable charts have a minimum zoom factor. We’ll be looking into adding greater flexibility in this respect. The thing to consider here, however, is the number of major entries, not the range of _x_ values covered. Proportional positioning doesn’t necessarily mean the content will be wider. To scale everything down horizontally, you can use a custom _x_ step, which is the difference between the _x_ values of neighboring major entries. You could set it to the number of milliseconds in a minute, hour, or day, for example. (Here, as initially, you’d use millisecond timestamps, not consecutive integers.)
